src/scripts/helpers/apis.mjs

```diff
@@ -0,0 +1,216 @@
+
+    return executeData;
+}
+
+/**
+ * 
+ * @param {string} endpoint 
+ * @param {string} method 
+ * @param {object} data 
+ * @returns 
+ */
+export const doSecureFetch = async(endpoint, method, data=null) => {
+    if(method == 'GET') {
+
+        const executeFetch = await fetch(endpoint, {
+            method: method,
+            headers: {
+                Authorization: `Bearer ${storedUser.accessToken}`,
+            },
+        });
+
+        const executeData = await executeFetch.json();
+
+        return executeData;
+
+    } else if(method == 'POST') {
+
+
+        const executeFetch = await fetch(endpoint, {
+            method: method,
+            body: JSON.stringify(data),
+            headers: {
+                'Content-type': 'application/json; charset=UTF-8',
+                Authorization: `Bearer ${storedUser.accessToken}`,
+            },
+        });
+
+        const executeData = await executeFetch.json();
+
+        return executeData;
+    } else {
+        return null;
+    }
+}
+
+/**
+ * 
+ * @param {*} otherUsername 
+ * @returns 
+ */
+export const fetchOthersPost = async(otherUsername) => {
+
+    const otherPostsEndpoint = baseUrl + `/social/profiles/${otherUsername}/posts`;
+    const executeFetch = await fetch(endpoint, {
+        method: method,
+        headers: {
+            Authorization: `Bearer ${storedUser.accessToken}`,
+        },
+    });
+
+    const executeData = await executeFetch.json();
+
+    return executeData;
+}
+
+/**
+ * 
+ * @returns 
+ */
+export const getCurrentUserPosts = async() => {
+    const getMyPostsEndpoint = baseUrl + `/social/profiles/${storedUser.profileName}/posts?&_active=true&_author=true&_comments=true&_reactions=true`;
+
+    const executeFetch = await fetch(getMyPostsEndpoint, {
+        method: 'GET',
+        headers: {
+            Authorization: `Bearer ${storedUser.accessToken}`,
+        },
+    });
+
+    const executeData = await executeFetch.json();
+
+    return executeData;
+}
+
+
+export const deletePostByPostId = async(postId) => {
+   
+    const deletePostEndpoint = baseUrl + `/social/posts/${postId}`;
+
+    const executeFetch = await fetch(deletePostEndpoint, {
+        method: 'DELETE',
+        headers: {
+            Authorization: `Bearer ${storedUser.accessToken}`,
+        },
+    });
+
+    const executeData = await executeFetch.json();
+
+    return executeData;
+};
+
+export const getPostById = async(id) => {
+
+    const getSingleEntryEndpoint = baseUrl + `/social/posts/${id}?&_author=true`;
+
+    const executeFetch = await fetch(getSingleEntryEndpoint, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${storedUser.accessToken}`,
+      },
+    });
+
+    const executeData = await executeFetch.json();
+
+    return executeData;
+
+};
+
+
+export const updatePost = async (id, data) => {
+  const updateEndpoint = baseUrl + `/social/posts/${id}`;
+
+  console.log('data pass is ', data);
```

Comment:
```suggestion

```
You should remove all unnecessary console.log statements from your code. You can leave the ones that make sense, for example in the catch block of a try/catch statement.

Reply:
Thank you